src/modules/tas_studio/editor/db.rs

```diff
@@ -35,6 +47,278 @@ impl fmt::Debug for Branch {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SplitInfo {
+    // must guarantee to have framebulk before and after this
+    /// range of split marker
+    /// - starts from 1 line after previous framebulk
+    /// - ends before next framebulk
+    pub split_range: Range<usize>,
+    // for the sake of easier searching, the last framebulk index is stored
+    pub bulk_idx: usize,
+    // TODO: could probably get away with &str
+    pub name: String,
+    pub split_type: SplitType,
+    // ready as in, there's a save created, and lines before and including start_idx is still unchanged
+    pub ready: bool,
+    // TODO: timing on grabbing rng values
+    // TODO: how do i get shared rng from engine?
+    // TODO: i most likely need to add this information being sent from a sim client, how can this be done in a clean way
```

Comment:
I check `SplitInfo` for ready state inside `apply_accurate_frame`, I need shared and non-shared RNG so splits can use those information
Is it fine to just give those information in `AccurateFrame` struct? I don't know the alternate way to grab those information from whatever is sending `AccurateFrame`